src/endpoint/s3/s3_bucket_policy_utils.js

```diff
@@ -269,7 +269,12 @@ async function validate_s3_policy(policy, bucket_name, get_account_handler) {
         }
         for (const resource of _.flatten([statement.Resource || statement.NotResource])) {
             const resource_bucket_part = resource.split('/')[0];
-            const resource_regex = RegExp(`^${resource_bucket_part.replace(qm_regex, '.?').replace(ar_regex, '.*')}$`);
+            const resource_regex = RegExp(
+                `^${resource_bucket_part
+                .replace(/[-/^$+?.()|[\]{}]/g, '\\$&')
```

Comment:
notice that we are modifying the ? character right after your change. did you check it doesn't effect this? worth adding a test as well if one doesn't already exists

Reply:
@nadavMiz I think the question mark (?) will be replaced with `\?` in the 1st part then \? will be replace by `\.?` . Have limited or no knowledge on this. Checking this